src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriversCountVisitor.java

```diff
@@ -1,43 +1,46 @@
 package edu.kis.powp.jobs2d.drivers.visitor;
 
-import edu.kis.powp.jobs2d.Job2dDriver;
-import edu.kis.powp.jobs2d.drivers.DriverComposite;
-import edu.kis.powp.jobs2d.drivers.LineSimulatorDriver;
+import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
 import edu.kis.powp.jobs2d.drivers.adapter.DriverDecorator;
+import edu.kis.powp.jobs2d.drivers.composite.IDriverComposite;
 import edu.kis.powp.jobs2d.drivers.logger.LoggerDriver;
 
-import java.util.logging.Logger;
+import java.util.List;
+import java.util.stream.Collectors;
 
-public class DriversCountVisitor implements IVisitor {
+public class DriversCountVisitor implements IDriverVisitor {
     private long count = 0;
 
-    public long getDriversCount(){
-        return count;
+    public long getDriversCount() {
+        long tmpCount = count;
+        count = 0;
+        return tmpCount;
     }
 
     @Override
-    public void visit(DriverComposite driver) {
-        count += driver.getDrivers().size();
+    public void visit(IDriverComposite driver) {
+        List<VisitableJob2dDriver> drivers = driver.getDrivers().stream()
+                .map(driver1 -> (VisitableJob2dDriver) driver1)
+                .collect(Collectors.toList());
+
+        drivers.forEach(driver1 -> driver1.accept(this));
     }
 
     @Override
     public void visit(DriverDecorator driver) {
+        VisitableJob2dDriver wrappedDriver = (VisitableJob2dDriver) driver.getDriver();
+        wrappedDriver.accept(this);
         count++;
```

Comment:
OK, but when you visit DriverDecorator you increment count, but you don't do it for CompositeDriver. However, maybe that makes sense as decorators introduce additional responsibilities.